src/services/formatting/formatting.ts

```diff
@@ -605,22 +609,62 @@ namespace ts.formatting {
                     return inheritedIndentation;
                 }
 
-                if (isToken(child)) {
+                // inherit indentation of preceding argument
+                let {indentationLocked} = parentListStates || <ListStates>{};
+
+                if (isToken(child) && !outOfTargetRange) {
+                    // replace indentation with inherited one, ignoring delta to prevent unexpected over-indentation
+                    if (indentationLocked) {
+                        parentDynamicIndentation = getDynamicIndentation(parent, parentStartLine, inheritedIndentation, 0);
+                    }
+
                     // if child node is a token, it does not impact indentation, proceed it using parent indentation scope rules
                     let tokenInfo = formattingScanner.readTokenInfo(child);
                     Debug.assert(tokenInfo.token.end === child.end);
                     consumeTokenAndAdvanceScanner(tokenInfo, node, parentDynamicIndentation);
-                    return inheritedIndentation;
+                    if (checkListElementLockIndentation()) {
+                        parentListStates.indentationLocked = true;

```

Comment:
Above you check for `parentListStates` being defined; but can't it be undefined here?